Generate stable bindata output

liggitt opened the pull request:
fixes #30621

bindata generation produces spurious diffs when run with different go versions because gzip output is not stable between versions.

It also produces spurious diffs when invoked directly vs via make/go:generate.

This PR:
* adds data to bindata uncompressed (makes translation files human-readable, and the zip files were already compressed... for reference the file was 27.7 KB before, and 20.1 KB with this change)
* runs the generation from KUBE_ROOT, so it doesn't matter whether it is invoked directly or via go:generate

supercedes https://github.com/kubernetes/kubernetes/pull/39631

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/39673)
<!-- Reviewable:end -->

liggitt committed:
Generate translation bindata file without compression

gzip output is not stable between go releases, so bindata.go shows spurious diffs when compressed

also, half the output is human readable, and the other half is already zip compressed

deads2k commented:
This worked for me.

deads2k commented on hack/generate-bindata.sh:
```diff
@@ -63,7 +63,7 @@ rm -f "${BINDATA_OUTPUT}.tmp"
 # These are files for runtime code
 BINDATA_OUTPUT="${KUBE_ROOT}/pkg/generated/bindata.go"
-go-bindata -nometadata -prefix "${KUBE_ROOT}" -o "${BINDATA_OUTPUT}.tmp" -pkg generated \
+go-bindata -nometadata -nocompress -prefix "${KUBE_ROOT}" -o "${BINDATA_OUTPUT}.tmp" -pkg generated \
```
You ok doing this?  It makes my life a lot easier, so I can definitely live with it

liggitt replied:
I prefer it, actually... half the things being added are useful if they are human readable, and the other half are already zip-compressed

liggitt committed:
Run bindata generation from KUBE_ROOT

Avoid path differences when running via go:generate and invoking directly

liggitt commented:
fixed the difference between direct invocation and invocation via go:generate by doing pushd/popd and running the guts of the bindata generation from KUBE_ROOT

liggitt commented:
openapi flake fixed.
@k8s-bot unit test this

fabianofranz commented:
This looks better with `-nocompress` than without it. LGTM but I'll defer to @brendandburns.

liggitt commented:
rebased
